shell/components/nav/TopLevelMenu.vue

```diff
@@ -218,7 +218,9 @@ export default {
           params: { cluster }
         };
 
-        if ( !this.$router.getMatchedComponents(to).length ) {
```

Comment:
`getMatchedComponents` has been removed in vue3, so it's nice to get rid of -  

However they suggest [here](https://router.vuejs.org/guide/migration/#Removal-of-router-getMatchedComponents-) to check the route itself which seems nice. So something like

```
const pRoute = this.$router.resolve(to);
const matched = pRoute.resolved?.matched?.flatMap((record) => Object.values(record.components));
```

This would also catch cases where routes with a common name but different params are treated the same via the current check of `.getRoutes().filter((route) => route.name === to.name);`. Not sure how common that is, but would be good to handle

Reply:
Agree, I've verified that  
```
pRoute.resolved?.matched?.flatMap((record) => Object.values(record.components));
```
returns the same objects of `getMatchedComponents`.
Thanks for the heads up :+1:

Reply:
Well, `this.$router.resolve(to)` throws again the warning message.

Reply:
Hmm, it looks like anything with match will trigger it - https://github.com/vuejs/vue-router/blob/779dc5c716f82f097b48536037c4c1d5b7505339/src/create-matcher.js#L62

Reply:
Ok... it looks like matcher just uses name! Maybe we go back to the original solution and test for any change in side nav

Reply:
I've reverted the last commit.

Reply:
No issues :smiley: